src/gpu/d3d12/SDL_gpu_d3d12.c

```diff
@@ -2673,13 +2674,21 @@ static D3D12Texture *D3D12_INTERNAL_CreateTexture(
 
     heapFlags = D3D12_HEAP_FLAG_NONE;
 
+    if (((textureCreateInfo->depth <= 1) && (!textureCreateInfo->isCube) && (textureCreateInfo->layerCount > UINT16_MAX))
+        || ((textureCreateInfo->depth > 1) && (textureCreateInfo->depth > UINT16_MAX))
+        || (textureCreateInfo->levelCount > UINT16_MAX)) {
+        SDL_LogError(SDL_LOG_CATEGORY_GPU, "Failed to create texture! Argument out of range.");
+        D3D12_INTERNAL_DestroyTexture(renderer, texture);
+        return NULL;
+    }
```

Comment:
This part looks good, can we put it in gpu.c possibly?

Reply:
we could change levelCount/layerCount to UINT16 to avoid the possibility too.